src/kernel/idt.c

```diff
@@ -0,0 +1,15 @@
+#include "idt.h"
+
+void set_idt_gate(int index, uint32_t offset) {
+	idt[index].low_offset = offset & 0xffff;
+	idt[index].selector = GDT_KERNEL_CODE_SEG;
+	idt[index].zero = 0;
+	idt[index].attributes = 0x8e;
```

Comment:
This is not Hogwarts. Magic isn't allowed here.

Move to define pls

Reply:
yer a wizard harry, fixed